tests/unit/array/mixins/test_find.py

```diff
@@ -192,3 +193,42 @@ def test_find_by_tag(storage, config, start_storage):
     assert isinstance(results, list)
     assert len(results) == 2
     assert all([isinstance(result, DocumentArray) for result in results]) == True
+
+
+numeric_operators = {
+    '$gte': operator.ge,
+    '$gt': operator.gt,
+    '$lte': operator.le,
+    '$lt': operator.lt,
+    '$eq': operator.eq,
+    '$neq': operator.ne,
+}
+
+
+@pytest.mark.parametrize('operator', list(numeric_operators.keys()))
+def test_search_annlite_filter(tmpdir, operator):
+
+    Nq = 5
+    D = 128
+    columns = [('price', float), ('category', str)]
+    da = DocumentArray(
+        storage='annlite',
+        config={'n_dim': D, 'columns': columns, 'data_path': str(tmpdir)},
+    )
+
+    X = np.random.random((Nq, D)).astype(np.float32)
+    query_da = DocumentArray([Document(embedding=X[i]) for i in range(Nq)])
+
+    thresholds = [20, 50, 100, 400]
+
+    for threshold in thresholds:
+        da.find(
+            query_da, filter={'price': {operator: threshold}}, include_metadata=True
+        )
+        for query in query_da:
+            assert all(
+                [
```

Comment:
you should first assert the length of matches, otherwise this could be empty and pass

Reply:
updated with your check